### PR TITLE
feat: implement collision detection

### LINKED ISSUE #5: Implement collision detection

### PR DESCRIPTION
**Pull request by arsham**
Close #5

### CODE REVIEW

File: internal/quadtree/quadtree.go [hunk 1]
```diff
@@ -0,0 +1,238 @@
+package quadtree
+
```
Comment (daniel-np) [inline internal/quadtree/quadtree.go:4]: I see the linter is complaining about comments here.

Comment (arsham) [reply internal/quadtree/quadtree.go:4]: I will fix it, thanks!

File: internal/scene/play.go [hunk 2]
```diff
@@ -40,6 +41,8 @@ func NewPlay(c controller) *Play {
 	p.registerAction(ebiten.KeyB, action.ToggleBoundingBoxes)
 	p.registerAction(ebiten.KeySpace, action.Pause)
 	p.registerAction(ebiten.KeyT, action.ToggleTextures)
+	p.registerAction(ebiten.KeyC, action.ToggleCollisions)
+	p.registerAction(ebiten.KeyD, action.ToggleCollisionBoxes)
```
Comment (daniel-np) [inline internal/scene/play.go:45]: Is not the bounding box the collision box?

**Review by daniel-np — APPROVED** [review]

File: internal/scene/play.go (hunk 2 above)
Comment (arsham) [reply internal/scene/play.go:45]: It is different. The BB is useful if we want not to draw things but their area box. It doesn't mean they can have collision. Imagine we have entities for pollution on the map, they can't collide with anything but we still can draw their BB.

**Pull request merged**